scripts/unix/download-tool.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+###
+# Copyright 2016 resin.io
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+
+set -u
+set -e
+
+function check_dep() {
+  if ! command -v $1 2>/dev/null 1>&2; then
+    echo "Dependency missing: $1" 1>&2
+    exit 1
+  fi
+}
+
+check_dep wget
+
+if command -v sha256sum 2>/dev/null 1>&2; then
+  SHA256SUM=sha256sum
+elif command -v shasum 2>/dev/null 1>&2; then
+  SHA256SUM="shasum -a 256"
+else
+  echo "Dependency missing: sha256sum or shasum" 1>&2
+  exit 1
+fi
+
+function usage() {
+  echo "Usage: $0"
+  echo ""
+  echo "Options"
+  echo ""
+  echo "    -u <url>"
+  echo "    -c <sha256 checksum>"
+  echo "    -x set executable permissions"
+  echo "    -o <output>"
+  exit 1
+}
+
+ARGV_URL=""
+ARGV_CHECKSUM=""
+ARGV_EXECUTION_PERMISSIONS=false
```

Comment:
Only a minor nitpick, but could this be renamed to `ARGV_EXECUTABLE_PERMISSIONS` ?

Although looking at https://en.wikipedia.org/wiki/Chmod perhaps it's be better to rename this to `ARGV_EXECUTE_PERMISSION` ? (and update the usage text similarly)